torch/csrc/api/include/torch/nn/modules/transformer.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/nn/module.h>
+#include <torch/nn/options/transformer.h>
+#include <torch/nn/pimpl.h>
+#include <torch/nn/modules/activation.h>
+#include <torch/nn/modules/dropout.h>
+#include <torch/nn/modules/normalization.h>
+
+namespace torch {
+namespace nn {
+
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ TransformerDecoderLayer ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// TransformerDecoderLayer is made up of self-attn, multi-head-attn and feedforward network.
+/// This standard decoder layer is based on the paper "Attention Is All You Need".
+/// Ashish Vaswani, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan N Gomez,
+/// Lukasz Kaiser, and Illia Polosukhin. 2017. Attention is all you need. In Advances in
+/// Neural Information Processing Systems, pages 6000-6010. Users may modify or implement
+/// in a different way during application.
+/// See https://pytorch.org/docs/master/nn.html#transformer-layers to learn
+/// about the exact behavior of this module.
+///
+/// See the documentation for `torch::nn::TransformerDecoderLayerOptions` class to learn what
+/// constructor arguments are supported for this module.
+///
+/// Example:
+/// ```
+/// TransformerDecoderLayer model(TransformerDecoderLayerOptions(512, 8).dropout(0.2));
+/// ```
+class TORCH_API TransformerDecoderLayerImpl : public Cloneable<TransformerDecoderLayerImpl> {
+ public:
+  TransformerDecoderLayerImpl(int64_t d_model, int64_t nhead)
+    : TransformerDecoderLayerImpl(TransformerDecoderLayerOptions(d_model, nhead)) {}
+  explicit TransformerDecoderLayerImpl(const TransformerDecoderLayerOptions& options_);
+
+  void reset() override;
+
+  /// Pretty prints the `TransformerDecoderLayer` module into the given `stream`.
+  /// TODO: To be implemented along with the python API implementation
+  // void pretty_print(std::ostream& stream) const override;
```

Comment:
Please implement this method if having time. Basically, just printout all the submodules it has